### ISSUE TITLE
Poldercast update tracking issue

### ISSUE THREAD

**Issue by Zeegomo** [input-output-hk/jormungandr#3169]
- [x]  Trusted peers are not inserted in the topology right after startup anymore. Instead, we establish a connection and let them advertise themselves to the node. Basically that means that the node will need a bit more time to discover peers, and if a trusted peer is not available for gossip at startup we will forget about it and we will add it to the topology only after receiving a gossip about it. https://github.com/input-output-hk/jormungandr/pull/3133
- [x] Use `SocketAddr` instead of `Multiaddr` in p2p code and config for listen address. https://github.com/input-output-hk/jormungandr/pull/3133
- [x]  Re-implemented quarantine with `promote_peer` / `remove_peer`. https://github.com/input-output-hk/jormungandr/pull/3133
- [x] Removed old `legacy_node_id`. We can afford a breaking change in the config now and it probably conflicts with the new `node_key` field https://github.com/input-output-hk/jormungandr/pull/3133
- [x] Implement custom layer for preferred nodes (and maybe random direct connections?). It will not insert such nodes in the topology on its own but instead it will return a view based only on those nodes if they are already in the topology. https://github.com/input-output-hk/jormungandr/pull/3178
- [x] Implement custom ring layer to be able to specify per-topic interest as it is now https://github.com/input-output-hk/jormungandr/pull/3178
- [ ] There are a bunch of things that the new poldercast version does not support:
    - [x] It is not possible to implement custom layers since `PriorityMap` (type of one of the arguments of required methods for `Layer`) is not public. https://github.com/primetype/poldercast/pull/33
    - ~~[ ] Layers are trait objects and do not implement `Sync + Send` (currently using a fork, we might open a PR for that, or solve it using a task if that is possible)~~ Not needed anymore since we moved topology to its own task
    - [x] We do not have access to the inner nodes known to the topology. There are a bunch of methods used only by rest API IIRC that rely on this information. We could "reverse engineer" this info but it would be pretty inefficient because internally it uses a lru cache and we don't know when evictions happen. At the moment it just return the current view, so this is just a subset of known nodes. https://github.com/primetype/poldercast/pull/34
    - [x] We cannot retrieve our own profile from the topology. This is not too much of a problem for us since the node profile does not change with time we can simply save a copy before creating the topology, but it requires to duplicate a piece of data that we already have https://github.com/primetype/poldercast/pull/34
    - [ ] Poldercast does not allow to reset view layers. Should we keep this option or remove it?
- [x] Move topology to its own task (and possibly trim down the global state that needs to be passed around in networking) https://github.com/input-output-hk/jormungandr/pull/3168

**Future upgrades**:

- [ ] Actually check that the `node_id` for trusted_peers / preferred nodes is the one that we expect and that the ids coming from external nodes do not change over time

_Originally posted by @Zeegomo in https://github.com/input-output-hk/jormungandr/issues/3133#issuecomment-802685917_
